localstack/services/events/rule.py

```diff
@@ -182,5 +203,20 @@ def _check_target_limit_reached(self) -> bool:
             return True
         return False
 
+    def _validate_event_pattern(self, pattern):
```

Comment:
These checks are used in our v1 implementation since they are currently used by moto, I would keep them in since our goal is parity with v1.
As mentioned by you I would add a more sophisticated checking engine in a later stage.

Reply:
Since we decided to only use the JAVA rule matching engine, there is no parsing of the input pattern required anymore, therefore there is also no validation of the pattern required.